src/main/java/com/orange/lo/sample/mqtt2dis/MessageHandler.java

```diff
@@ -54,12 +54,12 @@ public void send() {
             while (!messageQueue.isEmpty()) {
                 messageBatch.add(messageQueue.poll());
                 if (messageBatch.size() == disProperties.getMessageBatchSize()) {
-                    disMessageSender.send(new ArrayList<>(messageBatch));
+                    disMessageSender.send(messageBatch);
```

Comment:
We cannot pass original array "messageBatch" as parameter to "send" method because in line 58 we clear this array so in  disMessageSender object this array will be empty so we need to pass copy of this array